oryx-app-mllib/src/main/java/com/cloudera/oryx/app/mllib/kmeans/KMeansEvaluation.java

```diff
@@ -166,11 +165,20 @@
       overallSilhouetteCoefficientForClustering += silhouetteCoefficientForCluster;
     }
 
-    double silhouetteCoefficient = overallSilhouetteCoefficientForClustering / data.count();
+    double silhouetteCoefficient = overallSilhouetteCoefficientForClustering / dataSize;
```

Comment:
Looking good except, shouldn't this reflect the sample size rather than the original data set?